src/Symfony/Component/Filesystem/Filesystem.php

```diff
@@ -694,6 +694,25 @@ public function appendToFile(string $filename, $content, bool $lock = false): vo
         }
     }
 
+    /**
+     * Returns the content of a file as a string.
+     *
+     * @throws IOException If the file cannot be read
+     */
+    public function readFile(string $filename): string
+    {
+        if (is_dir($filename)) {
+            throw new IOException(sprintf('Failed to read file "%s": File is a directory.', $filename));
+        }
```

Comment:
Hi! The idea and implementation are great :smiley: 

Would there be any point, in `src/Symfony/Component/Filesystem/Filesystem::readFile`, in checking whether the file is readable, as in [`src/Symfony/Component/Config/Util/XmlUtils::loadFile()`](https://github.com/symfony/symfony/blob/0019010afabb189ecf220c5c05c4b1e45d615acf/src/Symfony/Component/Config/Util/XmlUtils.php#L122)?

```suggestion
        if (is_dir($filename)) {
            throw new IOException(sprintf('Failed to read file "%s": File is a directory.', $filename));
        }
        
        if (is_readable($filename)) {
            throw new IOException(sprintf('Failed to read file "%s": File is not readable.', $filename));
        }
```

Reply:
If the file is not readable (for whatever reason), `file_get_contents()` triggers a meaningful warning that my implementation escalates to an exception already.

Reply:
Indeed  !

To test the idea to the limit (having a specific message if the file is not readable), I did a simple test:

```
-rw-r--r-- 1 root root   23 Mar  8 00:00 readable.md
--w------- 1 root root   23 Mar  8 00:00 not-readable.md
```

With `not-readable.md`, we have the following exception with the current version of the implementation:

```
IOException

Failed to read file "/.../not-readable.md": file_get_contents(/.../not-readable.md): Failed to open stream: Permission denied
```

If I force an exception with `is_readable()` in the `readFile()`, I get some erratic behaviour. I have an exception at `clear:cache`:

```
$ bin/console cache:clear

In Filesystem.php line 709:                                                                                                                                                                        
Failed to read file "/.../var/cache/dev/App_KernelDevDebugContainer.xml.meta": File is not readable.  
```

Or with `link`:

```
$ php link ../pr_54173
PHP Fatal error:  Uncaught Symfony\Component\Filesystem\Exception\IOException: Failed to read file "/.../symfony/src/Symfony/Bundle/TwigBundle/composer.json": File is not readable. in /.../symfony/src/Symfony/Component/Filesystem/Filesystem.php:709
```

On https://www.php.net/manual/en/function.is-readable.php, it is noted that:

> The results of this function are cached. See clearstatcache() for more details.

In the `readFile()` method, I run `clearstatcache()` before, but still get the same exceptions :-1: 

Conclusion: in the `readFile()` method, everything works fine without `is_readable()` :+1: